src/content/reference/react/StrictMode.md

```diff
@@ -825,7 +827,19 @@ Without Strict Mode, it was easy to miss that your Effect needed cleanup. By run
 [Read more about implementing Effect cleanup.](/learn/synchronizing-with-effects#how-to-handle-the-effect-firing-twice-in-development)
 
 ---
+### Fixing bugs found by detaching and re-attaching DOM refs in development {/*fixing-bugs-found-by-cleaning-up-and-re-attaching-dom-refs-in-development*/}
 
+<Canary> 
+In canary and experimental channels, React will run an extra setup+cleanup cycle in development for DOM refs, much like it does for Effects. 
+
+For [`ref` callbacks](/reference/react-dom/components/common#ref-callback), React will call the callback function with the DOM node as its argument. It will then call the callback's [cleanup function](reference/react-dom/components/common#returns) before calling the `ref` callback function again with the DOM node as its argument.
+
+Similarly, React will detach DOM refs that were created via `useRef` by setting `ref.current` to `null` before setting it back to the DOM node.
```

Comment:
useRef is probably more common than callback refs. Maybe just flip it to put this statement before the line above?

Reply:
i was debating doing this, but that's a good justification - thanks